### ISSUE TITLE
Bottles published but revision not merged

### ISSUE THREAD

**Issue by fxcoudert** [Homebrew/homebrew-test-bot#880]
We have regularly cases where bottles are published but the corresponding revision is not merged to homebrew-core. The latest is `opencv` in https://github.com/Homebrew/homebrew-core/pull/124578

We're left in a state that the test-bot alone cannot recover from:

```
Error: docker://ghcr.io/homebrew/core/opencv:4.7.0_2 already exists!
```

Published [3 days ago](https://github.com/Homebrew/homebrew-core/pkgs/container/core%2Fopencv). But the corresponding revision was not committed to homebrew-core, so any revision bump will now fail on bottles publishing.

How can we make the bottles publishing atomic? I.e., ensure that bottles are published if and only if the revision is committed?

**Comment by MikeMcQuaid** [comment]
> How can we make the bottles publishing atomic? I.e., ensure that bottles are published if and only if the revision is committed?

I don't think we really can do this without changing our workflow (which we should probably do).

Eventually we really need to get to a "bottle commits are pushed to PRs and only actually published when they are merged into `master`".

The [public beta for the new merge queue feature](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) could help us here. You can have additional CI jobs which are only run when something is "queued for merge" which, in our case, could be the job that does the actual publishing and then merges it to `master` only when the publishing succeeds.

> But the corresponding revision was not committed to homebrew-core, so any revision bump will now fail on bottles publishing.

Zero blame here but: any ideas why this happened?

CC @Homebrew/maintainers @Homebrew/core for input and thoughts here.

**Comment by SMillerDev** [comment]
> Zero blame here but: any ideas why this happened?

I'm guessing GitHub packages outage

**Comment by p-linnane** [comment]
> I'm guessing GitHub packages outage

Can confirm this is what happened.
